cucumber_cpp/library/cucumber_expression/Ast.cpp

```diff
@@ -0,0 +1,143 @@
+#include "cucumber_cpp/library/cucumber_expression/Ast.hpp"
+#include <cctype>
+#include <map>
+#include <ranges>
+#include <stdexcept>
+#include <string>
+#include <utility>
+#include <variant>
+#include <vector>
+
+namespace cucumber_cpp::library::cucumber_expression
+{
+    namespace
+    {
+        const std::map<char, TokenType>& DemarcationMap()
+        {
+            static const std::map<char, TokenType> demarcationMap = {
+                { '/', TokenType::alternation },
+                { '{', TokenType::beginParameter },
+                { '}', TokenType::endParameter },
+                { '(', TokenType::beginOptional },
+                { ')', TokenType::endOptional },
+            };
+
+            return demarcationMap;
+        }
+
+        const std::map<TokenType, std::string>& TokenTypeMap()
+        {
+            static const std::map<TokenType, std::string> tokenTypeMap = {
+                { TokenType::alternation, "alternation" },
+                { TokenType::beginParameter, "beginParameter" },
+                { TokenType::endParameter, "endParameter" },
+                { TokenType::beginOptional, "beginOptional" },
+                { TokenType::endOptional, "endOptional" },
+            };
+            return tokenTypeMap;
+        }
+    }
+
+    std::string Node::Text()
+    {
+        return static_cast<const Node*>(this)->Text();
+    }
+
+    std::string Node::Text() const
+    {
+        if (std::holds_alternative<std::string>(children))
+            return std::get<std::string>(children);
+
+        if (!std::get<std::vector<Node>>(children).empty())
+        {
+            std::string text{ std::get<std::vector<Node>>(children).front().Text() };
+            for (const auto& child : std::get<std::vector<Node>>(children) | std::views::drop(1))
+                text += child.Text();
+            return text;
+        }
+
+        return {};
+    }
+
+    std::vector<Node>& Node::Children()
+    {
+        return std::get<std::vector<Node>>(children);
+    }
+
+    const std::vector<Node>& Node::Children() const
+    {
+        return std::get<std::vector<Node>>(children);
+    }
+
+    bool Token::IsEscapeCharacter(char ch)
+    {
+        return ch == '\\';
+    }
+
+    TokenType Token::TypeOf(char ch)
+    {
+        if (std::isspace(ch) != 0)
```

Comment:
@daantimmer Crashes here when the feature file has lines such as  `Given a ← tuple(4.3, -4.2, 3.1, 1.0)` .
ch is -30 and does not pass the assertions `c >= -1 && c <= 255`

Reply:
That would be unicode, and I am so far not accounting for any unicode. It might be added later, but its not on my current timeline

Reply:
Interesting, its not crashing here: https://godbolt.org/z/3raoqqPva

Reply:
@alextech I've added explicit acceptance tests: https://github.com/philips-software/amp-cucumber-cpp-runner/pull/180/commits/fbd84b61e87da75500aa3702447f8d87df50402d

This passes

Reply:
Was using msvc.
Tested it on linux setup and does work.
MSVC appears to be more strict on integer range validation prior to calling actual method.
when I replaced this line with

 `if (std::isspace(static_cast<unsigned char>(ch)) != 0)` 

works on both platforms.

Reply:
@daantimmer Also tested with if (std::isspace(ch, std::locale()). Works too.